In [1]:
# Importing the required libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# Reading the cleaned dataset

data = pd.read_csv("Preporcessed_Plcement.csv")
data

sl_no  ssc_p  hsc_p  degree_p  etest_p  mba_p    salary gender    ssc_b  \
0      1.0  67.00  91.00     58.00     55.0  58.80  270000.0      M   Others   
1      2.0  79.33  78.33     77.48     86.5  66.28  200000.0      M  Central   
2      3.0  65.00  68.00     64.00     75.0  57.80  250000.0      M  Central   
3      4.0  56.00  52.00     52.00     66.0  59.43       0.0      M  Central   
4      5.0  85.80  73.60     73.30     96.8  55.50  425000.0      M  Central   
..     ...    ...    ...       ...      ...    ...       ...    ...      ...   
210  211.0  80.60  82.00     77.60     91.0  74.49  400000.0      M   Others   
211  212.0  58.00  60.00     72.00     74.0  53.62  275000.0      M   Others   
212  213.0  67.00  67.00     73.00     59.0  69.72  295000.0      M   Others   
213  214.0  74.00  66.00     58.00     70.0  60.23  204000.0      F   Others   
214  215.0  62.00  58.00     53.00     89.0  60.22       0.0      M  Central   

       hsc_b     hsc_s   degree_t workex specialisation      status  
0     Others  Commerce   Sci&Tech     No         Mkt&HR      Placed  
1     Others   Science   Sci&Tech    Yes        Mkt&Fin      Placed  
2    Central      Arts  Comm&Mgmt     No        Mkt&Fin      Placed  
3    Central   Science   Sci&Tech     No         Mkt&HR  Not Placed  
4    Central  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
..       ...       ...        ...    ...            ...         ...  
210   Others  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
211   Others   Science   Sci&Tech     No        Mkt&Fin      Placed  
212   Others  Commerce  Comm&Mgmt    Yes        Mkt&Fin      Placed  
213   Others  Commerce  Comm&Mgmt     No         Mkt&HR      Placed  
214   Others   Science  Comm&Mgmt     No         Mkt&HR  Not Placed  

[215 rows x 15 columns]

In [3]:
# Getting the dataset info

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
sl_no             215 non-null float64
ssc_p             215 non-null float64
hsc_p             215 non-null float64
degree_p          215 non-null float64
etest_p           215 non-null float64
mba_p             215 non-null float64
salary            215 non-null float64
gender            215 non-null object
ssc_b             215 non-null object
hsc_b             215 non-null object
hsc_s             215 non-null object
degree_t          215 non-null object
workex            215 non-null object
specialisation    215 non-null object
status            215 non-null object
dtypes: float64(7), object(8)
memory usage: 25.3+ KB


In [4]:
# Checking for null values

data.isna().sum()

sl_no             0
ssc_p             0
hsc_p             0
degree_p          0
etest_p           0
mba_p             0
salary            0
gender            0
ssc_b             0
hsc_b             0
hsc_s             0
degree_t          0
workex            0
specialisation    0
status            0
dtype: int64

# Finding the probability of similarity using T-Test

###   Unpaired T-Test ---- Independent sample
       Different group(Male, Female)
       Same Condition(Salary)

In [5]:
from scipy.stats import ttest_ind
data = data.dropna() 
Male = data[data['gender'] == 'M']['salary']
Female = data[data['gender'] == 'F']['salary']
ttest_ind(Male, Female)

Ttest_indResult(statistic=2.1103433958668765, pvalue=0.03599560646745178)

#### Result:
##### The T-Test result shows the probability of similarity as 3%  (less than 5%) which is acceptable 

###   Unpaired T-Test ---- Independent sample
       Different Group (Commerce, Science) 
       Same Condition (mba_p)

In [6]:
from scipy.stats import ttest_ind
Commerce = data[data['hsc_s'] == 'Commerce']['mba_p']
Science = data[data['hsc_s'] == 'Science']['mba_p']
ttest_ind(Commerce, Science)

Ttest_indResult(statistic=-0.7331285580404581, pvalue=0.46432995253854314)

#### Result:
##### The T-Test result shows the probability of similarity as 46% (Greater than 5%) which is  not an acceptable range.

###   Paired T-Test ---- Dependent sample
       Same group(Male)
       Different Condition(ssc_p, hsc_p)

In [7]:
from scipy.stats import ttest_rel
Male_1 = data[data['gender'] == 'M']['ssc_p']
Male_2 = data[data['gender'] == 'M']['hsc_p']
ttest_rel(Male_1, Male_2)

Ttest_relResult(statistic=0.6852881050898078, pvalue=0.49431164600731503)

#### Result:
##### The T-Test result shows the probability of similarity as 49% (Greater than 5%) which is not an acceptable range.

###   Paired T-Test ---- Dependent sample
       Same group(Commerce)
       Different Condition(etest_p, mba_p)

In [8]:
from scipy.stats import ttest_rel
comm_1 = data[data['hsc_s'] == 'Commerce']['etest_p']
comm_2 = data[data['hsc_s'] == 'Commerce']['mba_p']
ttest_rel(comm_1, comm_2)

Ttest_relResult(statistic=7.868552092606869, pvalue=2.462926468454984e-12)

#### Result:
##### The T-Test result shows the probability of similarity as 246% (Greater than 5%) which is not an acceptable range.

# Hypothesis Testing

#### Null hypothesis (H0): This is the default assumption, usually stating that there is no effect or no difference.

#### Alternative hypothesis (H1 or Ha): This is the assertion you want to test, usually stating that there is an effect or a difference.

## Statement:1
     There is no difference between the SSC pass mark of Male And HSC pass mark of Male

In [9]:
from scipy.stats import ttest_rel
Male_1 = data[data['gender'] == 'M']['ssc_p']
Male_2 = data[data['gender'] == 'M']['hsc_p']
ttest_rel(Male_1, Male_2)

Ttest_relResult(statistic=0.6852881050898078, pvalue=0.49431164600731503)

# Report:
    As the P value is 0.494, Which is greater than the significant value of 0.05.
    We fail to reject the Null Hypothesis.
    In other words there is not enough evidence to support Alternate Hypothesis.

# Statement:2
    There is no difference between the salary of male and salary of female

In [10]:
from scipy.stats import ttest_ind
Male = data[data['gender'] == 'M']['salary']
Female = data[data['gender'] == 'F']['salary']
ttest_ind(Male, Female)

Ttest_indResult(statistic=2.1103433958668765, pvalue=0.03599560646745178)

# Report:
    As the P value is 0.0359, which is less than the significant value of 0.05.
    We reject the Null Hypothesis.
    In other words there is evidence to support Alternate Hypothesis.